src/vm/eventpipe.cpp

```diff
@@ -392,7 +389,7 @@ EventPipeSession *EventPipe::GetSession(EventPipeSessionID id)
     {
         CrstHolder _crst(GetLock());
 
-        if (!s_tracingInitialized)
+        if (s_state != EventPipeState::Initialized)
```

Comment:
Can you check if this is being called by the thread dispatching events to EventListeners? If it is I assume this API gets called after EventPipe::Shutdown has begun and it doesn't stop getting called until the EventListener session has been disabled. As written, both before and after your change it would trigger the assert. I suspect the correct check is `s_state < Initialized`

Reply:
I did not see it being called in my example, but I think it makes sense to update this condition.